src/core/src/main/java/org/apache/jmeter/reporters/ResultCollector.java

```diff
@@ -338,6 +353,22 @@ public void testStarted(String host) {
         if(summariser != null) {
             summariser.testStarted(host);
         }
+        errorOnlyCached = isErrorLogging();
+        successOnlyCached = isSuccessOnlyLogging();
+        fileNameCached = getFilename();
+    }
+
+    private void saveSampleEvent(SampleEvent event, SampleSaveConfiguration saveConfig) {
+        try {
+            event.getResult().setSaveConfig(saveConfig);
+            if (saveConfig.saveAsXml()) {
+                SaveService.saveSampleResult(event, out);
+            } else { // !saveAsXml
+                CSVSaveService.saveSampleResult(event, out);
+            }
+        } catch (Exception err) {
+            log.error("Error trying to record a sample", err); // should throw exception back to caller
```

Comment:
"should throw exception back to caller" this is confusing, it doesn't look like the code is doing this, is this a TODO or a mistake?

Reply:
This was already there ASIS so we didn't touch it